Tests/TuistKitTests/Services/InstallServiceTests.swift

```diff
@@ -169,4 +169,98 @@ final class InstallServiceTests: TuistUnitTestCase {
             update: false
         )
     }
+
+    func test_resolve_with_spm_arguments_from_config() async throws {
+        // Given
+        let stubbedPath = try temporaryPath()
+
+        given(manifestFilesLocator)
+            .locatePackageManifest(at: .any)
+            .willReturn(stubbedPath.appending(components: "Tuist", "Package.swift"))
+
+        let stubbedSwiftVersion = TSCUtility.Version(5, 3, 0)
+        given(configLoader)
+            .loadConfig(path: .any)
+            .willReturn(
+                Config.test(
+                    swiftVersion: .init(stringLiteral: stubbedSwiftVersion.description),
+                    installOptions: .test(
+                        passthroughSwiftPackageManagerArguments: ["--replace-scm-with-registry"]
+                    )
+                )
+            )
+
+        pluginService.fetchRemotePluginsStub = { _ in
+            _ = Plugins.test()
+        }
+
+        try fileHandler.touch(
+            stubbedPath.appending(
+                component: Manifest.package.fileName(stubbedPath)
+            )
+        )
+
+        var swiftPackageManagerControllerResolveArguments: [String]?
+        swiftPackageManagerController.resolveStub = { _, arguments, _ in
```

Comment:
It'd be amazing if the `swiftPackageManagerController` here would be using the `MockSwiftPackageManagerControlling` generated by [Mockable](https://github.com/Kolos65/Mockable). The `SwiftPackageManagerControlling` protocol will need the `@Mockable` macro.

You can find tons of other tests using this macro for inspiration.

Ideally, we'd use the opportunity here to rewrite the tests in this service and possibly remove the custom `MockSwiftPackageManagerController` altogether. If you don't feel comfortable doing that, I can do the cleanup in a follow-up PR.

Reply:
Sounds interesting, I will review the code using the `@Mockable` macro. I don’t feel comfortable rewriting the unit tests as a part of this PR, thanks for asking. If there’s anything needed in order to make it “ready to merge”, I try to handle it!